src/lib/operators/sort.cpp

```diff
@@ -202,8 +202,11 @@ class Sort::SortImpl : public AbstractReadOnlyOperatorImpl {
     auto materialization = std::make_shared<SortImplMaterializeOutput<SortColumnType>>(_table_in, _row_id_value_vector,
                                                                                        _output_chunk_size);
     auto output = materialization->execute();
-    for (auto& chunk : output->chunks()) {
-      chunk->set_ordered_by(std::make_pair(_column_id, _order_by_mode));
+
+    auto output_table = std::const_pointer_cast<Table>(output);  // cast away constness to call chunk->set_ordered_by
```

Comment:
Argh. No, this shouldn't be necessary. **const cast is evil.** Is it necessary for `SortImplMaterializeOutput` to return a const?

Reply:
Not really, it seems: I changed `SortImplMaterializeOutput` to return a non-const Table. Everything compiles and tests were fine.